lib/active_merchant/billing/gateways/trans_first_transaction_express.rb

```diff
@@ -0,0 +1,574 @@
+          add_amount(doc, amount)
+          add_original_transaction_data(doc, transaction_id)
+        end
+
+        commit(:capture, request)
+      end
+
+      def void(authorization, options={})
+        action, transaction_id = split_authorization(authorization)
+
+        request = build_xml_transaction_request do |doc|
+          add_original_transaction_data(doc, transaction_id)
+        end
+
+        commit(void_type(action), request)
+      end
+
+      def refund(amount, authorization, options={})
+        action, transaction_id = split_authorization(authorization)
+
+        request = build_xml_transaction_request do |doc|
+          add_amount(doc, amount)
+          add_original_transaction_data(doc, transaction_id)
+          add_order_number(doc, options)
+        end
+
+        commit(:refund, request)
+      end
+
+      def credit(amount, payment_method, options={})
+        request = build_xml_transaction_request do |doc|
+          add_pan(doc, payment_method)
+          add_amount(doc, amount)
+        end
+
+        commit(:credit, request)
+      end
+
+      def verify(credit_card, options={})
+        request = build_xml_transaction_request do |doc|
+          add_payment_method(doc, credit_card)
+          add_contact(doc, credit_card.name, options)
+        end
+
+        commit(:verify, request)
+      end
+
+      def store(payment_method, options={})
+        store_customer_request = build_xml_payment_storage_request do |doc|
+          store_customer_details(doc, payment_method.name, options)
+        end
+
+        MultiResponse.run do |r|
+          r.process { commit(:store, store_customer_request) }
+          return r unless r.success? && r.params["custId"]
+          customer_id = r.params["custId"]
+
+          store_payment_method_request = build_xml_payment_storage_request do |doc|
+            doc["v1"].cust do
+              add_customer_id(doc, customer_id)
+              doc["v1"].pmt do
+                doc["v1"].type 0 # add
+                add_payment_method(doc, payment_method)
+              end
+            end
+          end
+
+          r.process { commit(:store, store_payment_method_request) }
+        end
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(%r((<[^>]+pan>)[^<]+(<))i, '\1[FILTERED]\2').
+          gsub(%r((<[^>]+sec>)[^<]+(<))i, '\1[FILTERED]\2').
+          gsub(%r((<[^>]+id>)[^<]+(<))i, '\1[FILTERED]\2').
+          gsub(%r((<[^>]+regKey>)[^<]+(<))i, '\1[FILTERED]\2')
+      end
+
+      private
+
+      CURRENCY_CODES = Hash.new{|h,k| raise ArgumentError.new("Unsupported currency: #{k}")}
+      CURRENCY_CODES["USD"] = "840"
+
+      V1_NAMESPACE = "http://postilion/realtime/merchantframework/xsd/v1/"
+      SOAPENV_NAMESPACE = "http://schemas.xmlsoap.org/soap/envelope/"
+      AUTHORIZATION_FIELD_SEPARATOR = "|"
+
+      APPROVAL_CODES = %w(00 10)
+
+      RESPONSE_MESSAGES = {
```

Comment:
Hah no problem. I actually just did a commit to move them down here because it seemed like a lot of noise before the actual code.